jwt implementation

ADIGrimm opened the pull request.

ADIGrimm committed:
jwt implementation

Elena-Bruyako commented on src/main/java/online/bookstore/controller/AuthenticationController.java:
```diff
@@ -15,10 +18,16 @@
     }
 
+    @PostMapping("/login")
+    public UserLoginResponseDto login(@RequestBody UserLoginRequestDto requestDto) {
```
```suggestion
    public UserLoginResponseDto login(@RequestBody @Valid UserLoginRequestDto requestDto) {
```

Elena-Bruyako commented on src/main/java/online/bookstore/controller/AuthenticationController.java:
```diff
@@ -15,10 +18,16 @@
         return userService.registration(requestDto);
     }
 
+    @PostMapping("/login")
```
Add Swagger annotations

Elena-Bruyako commented on src/main/java/online/bookstore/dto/user/UserLoginRequestDto.java:
```diff
@@ -0,0 +1,4 @@
+package online.bookstore.dto.user;
+
+public record UserLoginRequestDto(String email, String password) {
```
add validation

Elena-Bruyako commented on src/main/java/online/bookstore/security/JwtAuthenticationFilter.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
```suggestion
        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
```

Elena-Bruyako commented on src/main/java/online/bookstore/security/JwtAuthenticationFilter.java:
```diff
@@ -0,0 +1,50 @@
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
+            return bearerToken.substring(7);
```
```suggestion
            return bearerToken.substring(constant.length());
```

Elena-Bruyako commented on src/main/java/online/bookstore/security/JwtAuthenticationFilter.java:
```diff
@@ -0,0 +1,50 @@
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
create constant for `"Bearer "`

Elena-Bruyako commented on src/main/resources/application.properties:
```diff
@@ -6,3 +6,6 @@ spring.datasource.password=
 spring.jpa.show-sql=true
 spring.jpa.open-in-view=false
+
+jwt.expiration=300000
```
don't forget to add these properties to test/app.properties

Elena-Bruyako reviewed (changes requested):
see comments

ADIGrimm committed:
added validation to UserLoginRequestDto
added constant TOKEN_TYPE ("Bearer ") to JwtAuthenticationFilter
added swagger annotations to AuthenticationController

liliia-ponomarenko commented on src/main/java/online/bookstore/security/JwtAuthenticationFilter.java:
```diff
@@ -0,0 +1,52 @@
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    public static final String TOKEN_TYPE = "Bearer ";
```
```suggestion
    private static final String TOKEN_TYPE = "Bearer ";
```

liliia-ponomarenko reviewed (approved):
Great job!

ADIGrimm committed:
Changed am in JwtAuthenticationFilter constant
